httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java

```diff
@@ -260,35 +260,36 @@ public void cancelled() {
 
     @Test
     void testGetWithTimeout() {
+        final AtomicBoolean isFutureCompleted = new AtomicBoolean(false);
+
         final FutureCallback<String> callback = new FutureCallback<String>() {
             @Override
             public void completed(final String result) {
-                // Nothing to do
+                isFutureCompleted.set(true);
             }
 
             @Override
             public void failed(final Exception ex) {
-                // Nothing to do
+                // Nothing to do here for this example
             }
 
             @Override
             public void cancelled() {
-                // Nothing to do
+                // Nothing to do here for this example
             }
         };
 
         final BasicFuture<String> future = new BasicFuture<>(callback);
 
-        new Thread(() -> {
-            try {
-                Thread.sleep(100);  // This simulates the delay in completing the future.
-                future.completed("test");
-            } catch (final InterruptedException e) {
-                future.failed(e);
-            }
-        }).start();
+        new Thread(() -> future.completed("test")).start();
 
+        // Poll until the future is completed or timeout
         assertTimeoutPreemptively(Duration.ofMillis(200), () -> {
+            while (!isFutureCompleted.get()) {
```

Comment:
@arturobernalg Is busy loop the only way to reliably test this condition?

Reply:
@ok2c 
A busy loop (or spinning loop) is not the only way to reliably test asynchronous conditions. In fact, in many cases, it's not the most efficient or recommended approach due to its CPU-intensive nature

if you believe the busy loop approach is more suitable for our testing environment, we can implement it.

Reply:
Wouldn't a `CountDownLatch` work for this case?

Reply:
Yes, a `CountDownLatch` is indeed an alternative to achieve the same result.

Reply:
It's not an alternative.